tests/e2e/tidbcluster/across-kubernetes.go

```diff
@@ -240,6 +241,328 @@ var _ = ginkgo.Describe("[Across Kubernetes]", func() {
+			err = InstallXK8sTiDBComponentsCertificates(ns3, tcName3, clusterDomain)
+			framework.ExpectNoError(err, "failed to install tidb components certificates for cluster: %q", tcName3)
+
+			ginkgo.By("Creating x-k8s tidb clusters with TLS enabled")
+			tc1.Spec.TiDB.TLSClient = &v1alpha1.TiDBTLSClient{Enabled: true}
+			tc1.Spec.TLSCluster = &v1alpha1.TLSCluster{Enabled: true}
+			utiltc.MustCreateTCWithComponentsReady(genericCli, oa, tc1, 10*time.Minute, 30*time.Second)
+
+			tc2.Spec.TiDB.TLSClient = &v1alpha1.TiDBTLSClient{Enabled: true}
+			tc2.Spec.TLSCluster = &v1alpha1.TLSCluster{Enabled: true}
+			utiltc.MustCreateTCWithComponentsReady(genericCli, oa, tc2, 10*time.Minute, 30*time.Second)
+
+			tc3.Spec.TiDB.TLSClient = &v1alpha1.TiDBTLSClient{Enabled: true}
+			tc3.Spec.TLSCluster = &v1alpha1.TLSCluster{Enabled: true}
+			utiltc.MustCreateTCWithComponentsReady(genericCli, oa, tc3, 10*time.Minute, 30*time.Second)
+
+			ginkgo.By("Fail TiKV in cluster-1 by setting a wrong image")
+			local, err := cli.PingcapV1alpha1().TidbClusters(tc1.Namespace).Get(context.TODO(), tc1.Name, metav1.GetOptions{})
+			framework.ExpectNoError(err, "getting tidbcluster %s/%s", tc1.Namespace, tc1.Name)
+			local.Spec.TiKV.BaseImage = inexistentBaseImage
+			err = genericCli.Update(context.TODO(), local)
+			framework.ExpectNoError(err, "updating tikv with an inexistent image %q for %q", tc1.Spec.TiKV.BaseImage, tcName1)
+			// force operator to trigger an upgrade.
+			err = c.AppsV1().StatefulSets(ns1).Delete(context.TODO(), fmt.Sprintf("%s-tikv", tcName1), metav1.DeleteOptions{})
+			framework.ExpectNoError(err, "deleting sts of tikv for %q", tcName1)
+
+			ginkgo.By("Waiting for tikv pod to be unavailable")
+			err = utiltc.WaitForTidbClusterCondition(cli, tc1.Namespace, tc1.Name, time.Minute*5, func(tc *v1alpha1.TidbCluster) (bool, error) {
+				down := true
+				for _, store := range tc.Status.TiKV.Stores {
+					down = down && (store.State == "Disconnected" || store.State == v1alpha1.TiKVStateDown)
+				}
+				return down || len(tc.Status.TiKV.FailureStores) != 0, nil
+			})
+			framework.ExpectNoError(err, "waiting for tikv pod to be unavailable")
+
+			ginkgo.By("Restart other component pods and check cluster status")
+			podList, err := c.CoreV1().Pods(ns1).List(context.TODO(), metav1.ListOptions{})
+			framework.ExpectNoError(err, "list pods under namespace %q", ns1)
+			for _, pod := range podList.Items {
+				if !strings.Contains(pod.Name, "pd") && !strings.Contains(pod.Name, "tikv") {
+					err := c.CoreV1().Pods(ns1).Delete(context.TODO(), pod.Name, metav1.DeleteOptions{})
+					framework.ExpectNoError(err, "failed to delete pod %q", pod.Name)
+				}
+			}
+
+			componentsFilter := make(map[v1alpha1.MemberType]struct{}, 2)
+			componentsFilter[v1alpha1.TiKVMemberType] = struct{}{}
+			err = oa.WaitForTidbComponentsReady(tc1, componentsFilter, 5*time.Minute, 10*time.Second)
```

Comment:
Check `IsTidbClusterAvailable` here.

Reply:
As pd/tikvs are in crash loop, current tc is not expect to be ready (i.e. available).
Or did you mean check whether other tc `IsTidbClusterAvailable`?

Reply:
OK, for the other TC, the similar case should have been checked in the other cases where we delete the whole cluster, it's OK not to check here.

Reply:
Sorry, it should be `tidbIsTLSEnabled`.